### DISCUSSION TITLE
🎧 Kaizen 13 - Should we build a CDN? - Jan. 12, 2024

### DISCUSSION THREAD

**Discussion by gerhard** [thechangelog/changelog.com#485] [Kaizen]
![shields.io](https://img.shields.io/badge/Last%20updated%20on%20January%2012%2C%202024-PUBLISHED-blue?style=for-the-badge)

[🎧 This is now available as changelog.com/friends/26](https://changelog.com/friends/26)

<a href="https://changelog.com/friends/26" target="_blank"><img width="1508" alt="Screenshot 2024-01-13 at 07 56 55" src="https://github.com/thechangelog/changelog.com/assets/3342/02deb0f3-c63f-485d-8544-471e65ad1c0f"></a>

See all improvements in this 🐙 GitHub repository: [Kaizen 12 ... 13](https://github.com/thechangelog/changelog.com/compare/kaizen.12...kaizen.13)

---

This contains all things related to the next Kaizen episode.

Share your thoughts via comments below:
- Is there something in changelog.com that you, our dear listener, would like us to tackle?
- Is there something that **you** would like to help out with?
- Anything that we - @adamstac @jerodsanto - want to improve?

A rough timeline of what to expect (red line is today):

```mermaid
gantt
    title 🎧 Kaizen 13 timeline
    dateFormat  YYYY-MM-DD
    axisFormat %b %e, %Y

    section CONTRIBUTE
    💻 ALL CONTRIBUTIONS WELCOME 👕 :2023-10-21, 76d

    section RECORD

    🎙 Record :milestone, 2024-01-05
    ✂️ Edit :2024-01-05, 7d
    🚀 SHIP IT! :milestone, 2024-01-12

    section NEXT
    Kick-off Kaizen 14 :2024-01-12, 7d
```

---

### What changed? [Kaizen.12 ... Kaizen.13](https://github.com/thechangelog/changelog.com/compare/kaizen.12...master)
- #419 
- #483
- [Add auth to Fastly purge requests](https://github.com/thechangelog/changelog.com/commit/9fc92dea8441c2ebbc73f0f396a8f7ffdcf35ce2)
- [Changelog Beats](https://changelog.com/beats)
- [Remove `NotionUpdater`](https://github.com/thechangelog/changelog.com/commit/c55ee397f290e2ac6cb4ebf5c660f47b4a3019b3)
- [@changelog on TikTok](https://github.com/thechangelog/changelog.com/commit/b86a5f9839556bbf58c65e76328a0861f0e21ba0)
- #490
  - flyctl `0.1.134` 👈 this started it - `dagger_on_fly` workflow was failing
  - Ubuntu `22.04` LTS (Jammy Jellyfish) `2023-10-04`
  - Elixir `1.14.5`
    - 🗣 https://github.com/thechangelog/changelog.com/pull/493
  - Golang `1.20.12`
  - Node.js `20.10.0` (from `18`)
  - Erlang `26.2` (from `25`)
- #495
- #496
- #492 
  - includes 1Password Service Account integration
  - upgrades Postgres from `15.3` to `16.1`
- #500 

---

### What else is coming up?
- #486
- [Reduce 404s](https://ui.honeycomb.io/changelog/datasets/fastly/result/xaa4Q7U4Aaq)
- Experiment with [middleware.io](https://middleware.io/)
- [Elixir / Phoenix releases](https://github.com/thechangelog/changelog.com/blob/aade90a1bed7c70e947f0462197b888ec695fbcd/magefiles/image/production.go#L35)
- Clean-up
  - [multiple arguments to rm -fr](https://github.com/thechangelog/changelog.com/blob/aade90a1bed7c70e947f0462197b888ec695fbcd/magefiles/image/production.go#L51)
  - [upload legacy assets](https://github.com/thechangelog/changelog.com/blob/aade90a1bed7c70e947f0462197b888ec695fbcd/magefiles/image/production.go#L128)
  - [wp-content redirect](https://github.com/thechangelog/changelog.com/blob/aade90a1bed7c70e947f0462197b888ec695fbcd/magefiles/image/production.go#L129)
  - [Why do we need to start the app - and therefore require the DB - to upload static assets?](https://github.com/thechangelog/changelog.com/blob/aade90a1bed7c70e947f0462197b888ec695fbcd/magefiles/image/production.go#L136)
- Daggerize `action-slack-notify`

---

### This is a follow-up to
- https://github.com/thechangelog/changelog.com/discussions/480
- https://github.com/thechangelog/changelog.com/discussions/469
- https://github.com/thechangelog/changelog.com/discussions/452

**Comment by ajhalili2006** [comment]
Chiming in regarding Changelog Beats: Do you consider get your Spotify artist page verified (by claiming it)?[^1] Hopefully someone will get some of the beats in next year's Wrapped (just joking, but :crossed_fingers:).

[^1]: YouTube might be a different beast in terms of [Official Aritst channels](https://support.google.com/youtube/answer/7336634?hl=en-A2zm0-4H7V7h10v2z) verification process.

**Comment by jerodsanto** [comment]
Interesting idea. What does verification accomplish, @ajhalili2006?

**Comment by StructPoltergeist** [comment]
This may help with shopping for or determining the complexity of a Content Delivery Network:

Apache Traffic Control is an Open Source implementation of a Content Delivery Network. 

- https://github.com/apache/trafficcontrol

These tools may help with building a Content Delivery Network:

Traefik (pronounced traffic) is a modern HTTP reverse proxy and load balancer.

- https://github.com/traefik/traefik

Envoy Proxy.

- https://github.com/envoyproxy/envoy

Consul is a Service Mesh.

- https://github.com/hashicorp/consul

Istio is a Service Mesh.

- https://github.com/istio/istio

This may help with Cache Misses:

Ristretto is a fast, concurrent cache library built with a focus on performance and correctness. 

- https://github.com/dgraph-io/ristretto

Related to: Fastly Content Delivery Network && Cache Misses Issues

**Reply by gerhard** [reply]
That's a good shout-out! Will check it out, thank you 👍
